src/api/procedures/controllerTransfer.ts

```diff
@@ -46,6 +46,15 @@ export async function prepareControllerTransfer(
 
   const originPortfolioId = portfolioLikeToPortfolioId(originPortfolio);
 
+  const { did } = await context.getSigningIdentity();
+
+  if (did === originPortfolioId.did) {
+    throw new PolymeshError({
+      code: ErrorCode.UnmetPrerequisite,
+      message: 'Transfers to self are not allowed',
```

Comment:
I'd change this to "Controller transfers to self are not allowed" just to be super explicit. Also, to avoid fetching the signing identity twice, you should add a `prepareStorage` method and fetch it there